### PR TITLE
Move Disable Teredo to Advanced and Add prefer-ipv4 tweak

### LINKED ISSUE #2614: Cant play overwatch since teredo is disabled. Cant reenable it
I cant enable teredo anymore and the restore point created doesnt enable it anymore

### PR DESCRIPTION
**Pull request by Marterich**
## Type of Change
- [x] New feature
- [x] Bug fix

## Description
- Move the "Disable Teredo" Tweak to advanced tweaks because of problems with XBOX Games and some other games that require the feature.
- Add a new tweak that prioritises IPv4 over IPv6 to get pretty much the same latency and security upsides without actually disabling the feature as an essential tweak

## Testing
Win11 Pro 23H2 VM

## Impact
Users should experience fewer unexpected problems in the future

## Issue related to PR

- Resolves #2614

## Additional Information
If I understand correctly, the dev docs links will be auto generated in the tweaks.json. If not, let me now or fix it yourself :) 

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: docs/dev/tweaks/Essential-Tweaks/Teredo.md
Comment (og-mrk) [inline docs/dev/tweaks/Essential-Tweaks/Teredo.md:1]: @Marterich can you run `devdocs-generator` script and submit the new documentation entries? it should result in the following changes:
- Moving `Teredo` Tweak from `Essential-Tweaks` to `z__Advanced Tweaks - CAUTION`
- Making a new entry in `Essential-Tweaks`, which's called `IPv46`

Thanks in advance, and as always great commits Marterich 😄

File: config/tweaks.json [hunk 1]
```diff
@@ -3162,7 +3179,7 @@
     "UndoScript": [
       "netsh interface teredo set state default"
     ],
-    "link": "https://christitustech.github.io/winutil/dev/tweaks/Essential-Tweaks/Teredo"
+    "link": "[auto generated if i understand #2481 corectly]"
```
Comment (og-mrk) [inline config/tweaks.json:3182]: You don't need to update it, as from what I understand, it'll be overwritten to the category & name of tweak, see [these lines](https://github.com/ChrisTitusTech/winutil/pull/2481/files#diff-ad8a830e5d482eefc3e300b32b2daa2c950fe09de54e0f66c2e4b86707913b98R534-R563) for more info, most notably is line 553, which's the value of link to use, hope this clears some things up 👍.

File: docs/dev/tweaks/Essential-Tweaks/Teredo.md
Comment (Marterich) [reply docs/dev/tweaks/Essential-Tweaks/Teredo.md:1]: All right, I ran and pushed the Docs Update. So this comment is resolved.
But I still have some thoughts. (If you want we can discuss this on discord or in an issue as well)
- Shouldn't this be automated (e.g on merge to main) 
- The devdocs-generator has to be optimized! when I ran it, it changed pretty much ALL devdoc files and not only the ones that I modified

Comment (og-mrk) [reply docs/dev/tweaks/Essential-Tweaks/Teredo.md:1]: @Marterich Thanks again for doing this, and yes I do agree that this should be automated, I was working on GitHub workflows for WinUtil ([link to PR](https://github.com/ChrisTitusTech/winutil/pull/2471)).. but sadly as time moved on (got busy with other things), I couldn't finish it.. plus it made a lot of changes, which I was planning to split it into multiple PRs, for easier testing and reviewing.

Also I do agree on optimizing the script, and at the time Drift & I (mostly Drift 😅, I only contributed on formatting/improving the script abit) wanted the script to just work.. but now it was tested (some bugs exist, like the ones you noticed [here](https://github.com/ChrisTitusTech/winutil/pull/2633/commits/197d777cf92dcb25935f0d506cd57b3c4b72caca) & [here](https://github.com/ChrisTitusTech/winutil/pull/2633/commits/5bf42269b8a18559bfe34883f8c7bff88d75023b), can be fixed with a simple patch), it's now considered stable, and we could start optimizing.

I've an idea on how to optimize it.. I just need to do it through a coding session, will update you about this over the Discord Server, plus the same idea could be applied to preprocessor script, which'll speed-up compile times by at least a couple hundred milliseconds.

**Comment by MyDrift-user** [comment]
@Marterich The link are autogenerated from devdocs-generator.ps1. 
Since it currently does not seem to be hooked up to any workflow it will not autogenerate.
I suggest running the autodocs generator again and then the link should be the correct one already.

**Comment by og-mrk** [comment]
> @Marterich The link are autogenerated from devdocs-generator.ps1. Since it currently does not seem to be hooked up to any workflow it will not autogenerate. I suggest running the autodocs generator again and then the link should be the correct one already.
> 
> EDIT: Note that this only has to be done because the script is not yet connected to a workflow

This was already addressed in [this review comment](https://github.com/ChrisTitusTech/winutil/pull/2633#discussion_r1733557529), and how it works was briefly explained in [this review comment](https://github.com/ChrisTitusTech/winutil/pull/2633#discussion_r1733560823), and yes.. currently the autodocs doesn't get run.. it should get ran after PR merges, which's better than after each commit done by PR author (in the Compile & test workflow), plus we can agree that the script is pretty stable, haven't notice any major problems with it, so committing changes made by the script is reasonable enough (at least IMO).

**Review by ChrisTitusTech — APPROVED** [review]

**Comment by ChrisTitusTech** [comment]
Thanks @Marterich I agree with this tweak as I have seen some downside to Teredo as well. In the future I might remove Teredo all together as I don't see a scenario where disabling it would be better than the prioritization method.

**Pull request merged**